.circleci/config.yml

```diff
@@ -260,32 +263,52 @@ workflows:
       - end2end_tests:
           requires:
             - install_dependencies
-      - aws-ecr/build-and-push-image:
-          name: Build and push
-          repo: "${ECR_REPO_NAME}"
-          tag: "${CIRCLE_SHA1}"
+      - build_and_push:
+          name: build_and_push_uat_image
+          image_type: "uat"
           requires:
             - install_dependencies
-      - aws-ecr/build-and-push-image:
-          name: Build and push latest tag
-          repo: "${ECR_REPO_NAME}"
-          tag: "latest"
+          filters:
+            branches:
+              ignore:
+                - main
+      - build_and_push:
+          name: build_and_push_production_image
+          image_type: "production"
           requires:
             - install_dependencies
           filters:
             branches:
               only:
                 - main
       - deploy:
-          name: Deploy UAT
+          name: Deploy branch UAT
```

Comment:
This repetition is distressing but I can't think of a way around it :(

Reply:
Ooh, I wonder if I can use the `$CIRCLE_BRANCH` variable instead of two branch filters...